File: android/src/main/java/com/amplitude/skylab/InMemoryStorage.java
package com.amplitude.skylab;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

class InMemoryStorage implements Storage {

    private final Map<String, Variant> data = new ConcurrentHashMap<>();

    @Override
    @NotNull
    public Variant put(@NotNull String key, @NotNull Variant value) {
        if (key != null) {
            return data.put(key, value);
        }
        return new Variant(null, null);
    }

    @Override
    @NotNull
    public Variant get(@NotNull String key) {
        if (key != null) {
            return data.get(key);
        }
        return new Variant(null, null);
    }

    @Override
    @NotNull
    public Map<String, Variant> getAll() {
        return new HashMap<>(data);
    }

    @Override
    public void clear() {
        data.clear();
    }
}

File: sdk/src/main/java/com/amplitude/experiment/storage/InMemoryStorage.kt
package com.amplitude.experiment.storage

import com.amplitude.experiment.Variant
import java.util.concurrent.ConcurrentHashMap

// test
internal class InMemoryStorage : Storage {

    private val data: MutableMap<String, Variant> = ConcurrentHashMap()

    override fun put(key: String, variant: Variant) {
        data[key] = variant
    }

    override operator fun get(key: String): Variant? {
        return data[key]
    }

    override fun remove(key: String) {
        data.remove(key)
    }

    override fun getAll(): Map<String, Variant> {
        return data.toMap()
    }

    override fun clear() {
        data.clear()
    }
}
